flink-clients/src/main/java/org/apache/flink/client/cli/CliFrontend.java

```diff
@@ -166,6 +166,10 @@ public Configuration getConfiguration() {
 		return copiedConfiguration;
 	}
 
+	public static Options getCustomCommandLineOptions() {
```

Comment:
so.....why does this getter have to be static?

Reply:
it should be possible for this getter to be non-static. Then the changes to the `customCommandLineOptions` field are unnecessary, and the in `FlinkShell` replace `CliFrontend.getCustomCommandLineOptions()` with `frontend.getCustomCommandLineOptions`